core/http-auth-aws/src/main/java/software/amazon/awssdk/http/auth/aws/signer/RegionSet.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.http.auth.aws.signer;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import software.amazon.awssdk.annotations.SdkPublicApi;
+import software.amazon.awssdk.utils.Validate;
+
+/**
+ * This class represents the concept of a set of regions.
+ * <p>
+ * A Re
```

Comment:
(javadoc incomplete)

What are the reasons we represent the values as a comma separated string instead of a `Set`? We discussed it would not be a plain String externally, but I thought we'd model the values separately. I may have missed something.

Reply:
It's not a plain string, but a wrapper of a string with validation upon creation, just as `RegionScope` is. `RegionSet` is just a more constrained version of `RegionScope`